src/basic/in-addr-util.c

```diff
@@ -464,3 +464,45 @@ int in_addr_mask(int family, union in_addr_union *addr, unsigned char prefixlen)
 
         return -EAFNOSUPPORT;
 }
+
+int in_addr_prefix_from_string(const char *p, int family, union in_addr_union *prefix, uint8_t *prefixlen) {
+        union in_addr_union buffer;
+        const char *e, *l;
+        uint8_t k;
+        int f, r;
+
+        assert(p);
+
+        e = strchr(p, '/');
+        if (e)
+                l = strndupa(p, e - p);
+        else
+                l = p;
+
+        r = in_addr_from_string_auto(l, &f, &buffer);
+        if (r < 0)
+                return r;
+
+        if (f != family)
+                return -1;
```

Comment:
-1 is not an OK error to return, see CODING_STYLE.

use -EINVAL or such.

That said, I am pretty sure this function should work much like in_addr_from_string_auto() and return which address family was actually parsed, accepting both AF_INET and AF_INET6... This check hence looks strange